### PR TITLE
Add python client SPIError

### PR DESCRIPTION
**Pull request by HaozhouWang**
1. Add SPIError exception for PL/Container python client
2. Now plpy SPI interface will throw exception and could be caught in PL Python code.
3. Add plpy.SPIError as new SPI exception object in python client.

### CODE REVIEW

**Comment by zhangh43** [comment]
try this case 
```
CREATE FUNCTION transfer_funds2() RETURNS void AS $$
try:
    with plpy.subtransaction():
        plpy.execute("UPDATE accounts SET balance = balance - 100 WHERE account_name = 'joe'")
        plpy.execute("UPDATE accounts SET balance = balance + 100 WHERE account_name = 'mary'")
except plpy.SPIError, e:
    result = "error transferring funds: %s" % e.args
else:
    result = "funds transferred correctly"
plan = plpy.prepare("INSERT INTO operations (result) VALUES ($1)", ["text"])
plpy.execute(plan, [result])
$$ LANGUAGE plpythonu;
```

File: src/pyclient/plpy_spi.c [hunk 1]
```diff
@@ -407,13 +438,13 @@ PLy_spi_execute_plan(PyObject *ob, PyObject *list, long limit) {
 	 * so if resp->cols > 0, it must be SELECT statment.
 	 */
 	if (resp->cols == 0) {
-		debug_print(WARNING, "the rows is %d", resp->rows);
+		debug_print(NOTICE, "the rows is %d", resp->rows);
```
Comment (paul-guo-) [inline src/pyclient/plpy_spi.c]: DEBUG1? debug_print->lprintf?

File: src/pyclient/plpy_spi.c [hunk 2]
```diff
@@ -685,13 +718,13 @@ PLy_spi_execute_query(char *query, long limit) {
...
```
Comment (paul-guo-) [inline src/pyclient/plpy_spi.c]: See previous comment.

File: tests/sql/spi_python.sql [hunk 3]
```diff
@@ -371,3 +371,16 @@ select pyspi_delete_notexist_execp();
...
+CREATE OR REPLACE FUNCTION pyspi_exec_exception() RETURNS integer AS $$
+# container: plc_python_shared
+plan = list();
+try:
+	rv = plpy.execute(plan);
+except Exception as e:
+	plpy.notice(type(e))
+return 0
+$$ LANGUAGE plcontainer;
+
```
Comment (paul-guo-) [inline tests/sql/spi_python.sql]: Need more test cases to cover code branches.

**Comment by HaozhouWang** [comment]
Tested this case, the error could not be caught by client as the type conversion error has been caught by QE side and not passed to client.

File: tests/sql/spi_python.sql (hunk 3 above)
Comment (HaozhouWang) [reply tests/sql/spi_python.sql]: Thx, fixed

**Pull request merged**